### PR TITLE
[New] support jest 27

### PR DESCRIPTION
**Pull request by jwir3**
This also upgrades dependency packages to avoid as many advisories as
possible. Note that some advisories were not able to be fixed, as the
dependencies including the respective package(s) have not yet been updated.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -126,7 +126,7 @@
 		"istanbul": "1.1.0-alpha.1",
 		"istanbul-lib-coverage": "^1.2.0",
 		"istanbul-merge": "^1.1.1",
-		"jest": "^18 || ^19 || ^20.0.1 || ^21 || ^22 || ^23 || ^24 || ^25 || ^26",
+		"jest": "^27.2.0",
```
Comment (ljharb) [inline package.json:129]: the dev dep range must exactly match the peer dep range.

File: package.json [hunk 2]
```diff
@@ -116,7 +116,7 @@
 		"semver": "^5.5.0"
 	},
 	"peerDependencies": {
-		"jest": "^18 || ^19 || ^20.0.1 || ^21 || ^22 || ^23 || ^24 || ^25 || ^26"
+		"jest": "^18 || ^19 || ^20.0.1 || ^21 || ^22 || ^23 || ^24 || ^25 || ^26 || ^27"
```
Comment (ljharb) [inline package.json:119]:
```suggestion
		"jest": "^18 || ^19 || ^20.0.1 || ^21 || ^22 || ^23 || ^24 || ^25 || ^26 || ^27.2.0"
```
might as well start with the latest v27?

**Review by ljharb — CHANGES_REQUESTED** [review]
this also would need to run tests on jest 27

File: package.json (hunk 1 above)
Comment (jwir3) [reply package.json:129]: Ok, I'll make this change.

File: package.json (hunk 2 above)
Comment (jwir3) [reply package.json:119]: Sounds good. I'll change it.

**Comment by jwir3** [comment]
> this also would need to run tests on jest 27

I'm not entirely sure how to make this change. Is it a change to CI configuration? I did run the tests using Jest 27 on my local machine, so I know they don't fail, but I'm not sure what is necessary here.

**Review by ljharb — APPROVED** [review]

**Comment by ljharb** [comment]
I'll rebase and land this after #25 is merged.

**Comment by jwir3** [comment]
Thank you!

**Pull request merged**